setup/setup_pyrobosim.bash

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# Sets up the pyrobosim virtual environment
+
+# User variables
+# Please modify these for your environment
+VIRTUALENV_FOLDER=~/python-virtualenvs/pyrobosim
+ROS_WORKSPACE=~/pyrobosim_ws
+
+# Create a Python virtual environment
+[ ! -d "${VIRTUALENV_FOLDER}" ] && mkdir -p ${VIRTUALENV_FOLDER}
+python3 -m venv ${VIRTUALENV_FOLDER}
+echo -e "Created Python virtual environment in ${VIRTUALENV_FOLDER}\n"
+
+# Install all the Python packages required
+# Note that these overlay over whatever ROS 2 already contains
+source "${VIRTUALENV_FOLDER}/bin/activate"
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+pushd "${SCRIPT_DIR}/.." > /dev/null
+pip3 install ./pyrobosim
+pip3 install -r test/python_test_requirements.txt
+
+# Write key variables to file
+ENV_FILE="pyrobosim.env"
+if [ -f "${ENV_FILE}" ]
+then
+  rm ${ENV_FILE}
+fi
+echo "# This is an autogenerated environment file for pyrobosim" >> ${ENV_FILE}
+echo "PYROBOSIM_VENV=${VIRTUALENV_FOLDER}" >> ${ENV_FILE}
+
+# If setting up with ROS, install extra packages
+echo -e ""
+read -p "Do you want to set up ROS? (y/n) : " USE_ROS
+if [ "${USE_ROS,,}" == "y" ]
+then
+  # Validate that the ROS workspace specified exists.
+  if [ ! -d "${ROS_WORKSPACE}" ]
+  then
+      echo -e "\nFolder '${ROS_WORKSPACE}' does not exist."
+      echo "Please configure it to the ROS workspace that contains the 'pyrobosim' repository."
+      return 1
```

Comment:
I think you want this to quit, right? This only works with source

```suggestion
      exit 1
```

Otherwise it doesn't actually quite (which is what I think you want?)

```
Folder '/home/parallels/ROS/deez/' does not exist.
Please configure it to the ROS workspace that contains the 'pyrobosim' repository.
./setup/setup_pyrobosim.bash: line 42: return: can only `return' from a function or sourced script
What ROS distro are you using? (humble, iron, jazzy, rolling) : ^C
```

Reply:
I see! 

So when there were `exit` commands next to the `pushd`/`popd`s and they failed, it would actually close the shell it ran in.

Will this happen here too? (I can also just try it later, but asking for general bash wisdom)

Reply:
Oh right - I believe it depends on how you run it. If you `source` it then everything runs in your current shell, so `exit` would do what you think it does. But with `./setup/setup_pyrobosim.bash` it runs in a new session and that would terminate this. I think you have that in your docs, but if we're supposed to source it that's fine too. Just update the setup instructions!

Reply:
Cool, thank you wise sir. I'll do some testing later and figure out what works best.

Reply:
Got it! Yeah, so it should be `exit` in the setup and `return` in the source script. Thank you wise sir.